repos/system_upgrade/el7toel8/actors/mariadbcheck/libraries/library.py

```diff
@@ -0,0 +1,227 @@
+from leapp.models import InstalledRedHatSignedRPM
+from leapp.libraries.common import reporting
+from leapp.libraries.stdlib import CalledProcessError, api, run
+from os import listdir
+from os.path import isfile, join, isdir
+import re
+
+
+def get_config_files(rootfile):
+    """ return MariaDB config files """
+
+    if not isfile(rootfile):
+        return []
+
+    files = [rootfile]
+    dirs = []
+    def get_new_files(directory):
+        new_configs = []
+        for f in listdir(directory):
+            path = join(directory, f)
+            if isfile(path) and path.endswith('.cnf') and path not in files:
+                new_configs.append(path)
+        return new_configs
+
+    def get_new_dirs(configfile):
+        values = []
+        with open(configfile, "r") as f:
+            m = re.findall(r"^\s*!includedir[\s=]+(.+)$" ,f.read(), flags=re.MULTILINE)
```

Comment:
Sorry if I'm not getting it, but won't m contain a list of 1 element all the time in case of a match?
If that's the case - maybe simplify L30-L33 removing the cycle?